adlfs/core.py

```diff
@@ -314,8 +314,13 @@ def mv(self, path1, path2):
     def concat(self, outfile, filelist):
         """ Concatenate a list of files into one new file"""
         outfile = AzureDLPath(outfile).trim()
-        self.azure.call('CONCAT', outfile.as_posix(),
-                        sources=','.join(filelist))
+        filelist = ','.join(AzureDLPath(f).as_posix() for f in filelist)
+        #self.azure.call('CONCAT', outfile.as_posix(),
+        #                sources=','.join(filelist))
+        #self.invalidate_cache(outfile)
+        #return
+        self.azure.call('MSCONCAT', outfile.as_posix(),
+                        data='sources='+filelist, deleteSourceDirectory='true')
```

Comment:
I would be careful setting this to always be true. The caller should be able to decide, since there could potentially be other files in the source directory they don't want deleted, and this will delete them. This optimization should really only be used by our uploader tools, and only if we can reasonably guarantee that the only files in the folder belong to the one target file being created. This goes back to my recommendation of creating unique folders specific to the file being uploaded, instead of using a single "tmp" folder for all uploads.